classes/Admin/Stats/Controller.php

```diff
@@ -0,0 +1,745 @@
+<?php
+declare( strict_types=1 );
+
+namespace Imagify\Admin\Stats;
+
+use Exception;
+use Imagify_DB;
+use WP_Date_Query;
+
+/**
+ * Admin stats processor.
+ */
+class Controller {
+
+	/**
+	 * Array of actions.
+	 *
+	 * @var array
+	 */
+	private static $actions = [
+		'imagify_count_attachments_as',
+		'imagify_count_error_attachments_as',
+		'imagify_count_optimized_attachments_as',
+		'imagify_count_saving_data_as',
+		'imagify_calculate_total_size_images_library_as',
+		'imagify_calculate_average_size_images_per_month_as',
+	];
+
+	/**
+	 * Stats option name.
+	 *
+	 * @var string
+	 */
+	protected $stats_option = 'imagify_admin_stats';
```

Comment:
I'm thinking this naming is not very accurate, what do you think about `imagify_attachments_stats` or `imagify_media_stats` instead?